studio/actions/manualTranslate.tsx

```diff
@@ -0,0 +1,166 @@
+import {SANITY_API_VERSION} from '../../config'
+
+type ModalFooterProps = {
+  onClose: () => void
+  onTranslate: () => void
+  disabled?: boolean
+}
+
+const ModalFooter = (props: ModalFooterProps) => {
+  const {onClose, onTranslate, disabled = false} = props
+  return (
+    <Flex width="100%" justify="flex-end">
+      <Inline space={2}>
+        <Button icon={CloseCircleIcon} onClick={onClose} text="Cancel" tone="critical" />
+        <Button
+          icon={TransferIcon}
+          onClick={onTranslate}
+          text="Translate"
+          tone="primary"
+          disabled={disabled}
+        />
+      </Inline>
+    </Flex>
+  )
+}
+
+type ModalContentProps = {
+  locales: string[]
+  disabled?: boolean
+  handleLocaleChange: (event: ChangeEvent<HTMLSelectElement>) => void
+}
+
+const ModalContent = (props: ModalContentProps) => {
+  const {locales, disabled = false, handleLocaleChange} = props
+  return (
+    <Stack space={4}>
+      <Stack space={3}>
+        <Text muted size={1}>
+          This document has already been translated for the following locales:
+        </Text>
+        <Inline space={2}>
+          {locales.map((locale) => {
+            return (
+              <Badge key={locale} tone="primary">
+                {locale}
+              </Badge>
+            )
+          })}
+        </Inline>
+      </Stack>
+      <Stack space={3}>
+        <Stack space={2}>
+          <Text size={1} weight="semibold">
+            Select Locale
+          </Text>
+          <Text size={1} muted>
+            Choose which locale you wish to translate this document to
+          </Text>
+        </Stack>
+        <Select disabled={disabled} onChange={handleLocaleChange}>
+          <option value="">Select a locale</option>
+          {Object.keys(LOCALES)
+            .filter((locale) => !locales.includes(locale))
+            .map((locale) => {
+              return (
+                <option key={locale} value={locale}>
+                  {LOCALES[locale].name}
+                </option>
+              )
+            })}
+        </Select>
+      </Stack>
+    </Stack>
+  )
+}
+
+export const ManualTranslateAction = (props: DocumentActionProps): DocumentActionDescription => {
+  const {id, type, published, draft} = props
+  const doc = draft || published
+  const toast = useToast()
+  const [modalOpen, setModalOpen] = useState(false)
+  const [locale, setLocale] = useState('')
+  const {data, loading, error} = usePostLocalesQuery({postId: id})
+  const client = useClient({
+    apiVersion: SANITY_API_VERSION,
+  })
+
+  const onClose = useCallback(() => {
+    setModalOpen(false)
+  }, [])
+
+  const onLocaleChange = useCallback((event: ChangeEvent<HTMLSelectElement>) => {
+    const locale = event.target.value
+    setLocale(locale)
+  }, [])
+
+  const sendTranslation = (locale: string) => {
+    return fetch('http://localhost:3000/api/translate', {
+      method: 'POST',
```

Comment:
❓ I'm assuming this is fine for now?

Reply:
For now yes, but well probably need to to adjust how this custom action works once the locale work has been merged. Is this ok with you?